packages/base-types/src/request/index.ts

```diff
@@ -68,16 +71,19 @@ export interface IntentRequestPayload extends ActionAndLabelRequestPayload {
   data?: Record<string, unknown>;
 }
 
-export interface IntentRequest extends BaseRequest<IntentRequestPayload> {
+export interface IntentRequest extends BaseRequest {
   type: RequestType.INTENT;
+  payload: IntentRequestPayload;
 }
 
-export interface GeneralRequest extends BaseRequest<ActionAndLabelRequestPayload> {
-  type: string; // the general request type is dynamic, used to m
+export interface GeneralRequest extends BaseRequest {
+  type: string; // the general request type is dynamic
+  payload: ActionAndLabelRequestPayload;
 }
 
-export interface ActionRequest extends BaseRequest<ActionAndLabelRequestPayload> {
+export interface ActionRequest extends BaseRequest {
   type: RequestType.ACTION;
+  payload: ActionAndLabelRequestPayload;
```

Comment:
All properties are optional for `ActionAndLabelRequestPayload`, so `payload` could be optional too.